Add support for visualizing bones in Advanced Import Settings

jeronimo-schreyer opened the pull request:
This commit adds support for visualizing bones in Advanced Import Settings. When a node with the `Skeleton3D` type is selected, the bones are drawn using lines or octahedrons to provide a clearer reference to their position.

Update
editor/import/scene_import_settings.cpp
editor/import/scene_import_settings.h

- *Bugsquad edit: This closes https://github.com/godotengine/godot-proposals/issues/6801.*

akien-mga commented:
> * _Bugsquad edit: This closes [Show skeleton bones in the Advanced Import Settings preview godot-proposals#6801](https://github.com/godotengine/godot-proposals/issues/6801)._

FYI "bugsquad edit" means am edit to another user's post by a member of Godot's bugsquad or bug triage team. So it doesn't make sense to use it in your own PRs ;)

jeronimo-schreyer commented:
Oh! I thought it was to connect it to a proposal in the other repo
sorry about that

TokageItLab commented on editor/import/scene_import_settings.cpp:
```diff
@@ -406,6 +406,134 @@ void SceneImportSettings::_fill_scene(Node *p_node, TreeItem *p_parent_item) {
+		Ref<ArrayMesh> m = surface_tool->commit();
+		m->surface_set_material(0, bones_preview_mesh_material);
+		bones_preview_mesh_instance->set_mesh(m);
+	}
```
Resend from: https://github.com/godotengine/godot-proposals/issues/6801#issuecomment-1538702093

> We have discussed about this and review https://github.com/godotengine/godot/pull/76818 with @lyuma, the main problem is that there is a lot of duplication of code from the SkeletonGizmo.

jeronimo-schreyer replied:
I'm thinking maybe a static function that returns the mesh representation of the bones (with the material already applied). Would that be ok?

TokageItLab replied:
Ah, yes. It might not be a great architectural design, but it certainly might be possible to avoid code duplication with static functions.

Probably I would not block that, but we need to hear the decisions of other contributors as well.

fire replied:
I think static functions is better than nothing, but I have no opinion.

jeronimo-schreyer replied:
I think we all agree static functions are awful 😁

Is there other way we could do this?

TokageItLab replied:
Well, it seems that no one is particularly opposed to that way, so for now I think it is fine to proceed with the implementation as static function.

fire replied:
Are all the reviews ready?

fire commented on editor/import/scene_import_settings.cpp:
```diff
@@ -406,6 +406,134 @@ void SceneImportSettings::_fill_scene(Node *p_node, TreeItem *p_parent_item) {
+				Vector3 d = (v1 - v0).normalized();
+				real_t dist = v0.distance_to(v1);
+
+				// Find closest axis.
```
Can you remove code that isn't used?

TokageItLab commented on editor/plugins/skeleton_3d_editor_plugin.cpp:
```diff
@@ -1363,6 +1335,39 @@ void Skeleton3DGizmoPlugin::redraw(EditorNode3DGizmo *p_gizmo) {
+	unselected_mat->set_flag(StandardMaterial3D::FLAG_ALBEDO_FROM_VERTEX_COLOR, true);
+	unselected_mat->set_flag(StandardMaterial3D::FLAG_SRGB_VERTEX_COLOR, true);
+
+	Ref<ShaderMaterial> selected_mat = Ref<ShaderMaterial>(memnew(ShaderMaterial));
+	Ref<Shader> selected_sh = Ref<Shader>(memnew(Shader));
```
The shaders must be stored somewhere since it is bad way to memnew them every time.

jeronimo-schreyer commented:
I added a struct to hold the materials but now is not passing the check

jeronimo-schreyer committed:
This commit adds support for visualizing bones in Advanced Import Settings.

When a node with the Skeleton3D type is selected, the bones are drawn using lines or octahedrons to provide a clearer reference to their position.
Refactored Skeleton3DGizmoPlugin::redraw now uses a static function to generate bone meshes

Update
editor/import/scene_import_settings.cpp
editor/import/scene_import_settings.h
editor/plugins/skeleton_3d_editor_plugin.h
editor/plugins/skeleton_3d_editor_plugin.cpp

fire commented:
It is failing here:

```c++
 #12 0x55e24ea7773e in Ref<ShaderMaterial>::~Ref() (/home/runner/work/godot/godot/bin/godot.linuxbsd.editor.dev.double.x86_64.san+0x47a0a73e)
    #13 0x55e2519078d0 in Skeleton3DGizmoPlugin::SelectionMaterials::~SelectionMaterials() (/home/runner/work/godot/godot/bin/godot.linuxbsd.editor.dev.double.x86_64.san+0x4a89a8d0)
```

jeronimo-schreyer commented:
oh oooooh...... since finally checks had passed I wanted to squash merge commits and my work..... I guess I kinda f** up right
